awswrangler/s3/_merge_upsert_table.py

```diff
@@ -47,22 +49,27 @@ def _is_data_quality_sufficient(
 ) -> bool:
     """Check data quality of existing table and the new delta feed."""
     error_messages = list()
+    existing_schema = pa.Schema.from_pandas(existing_df)
+    delta_schema = pa.Schema.from_pandas(delta_df)
     # Check for duplicates on the primary key in the existing table
     if sum(pandas.DataFrame(existing_df, columns=primary_key).duplicated()) != 0:
         error_messages.append("Data inside the existing table has duplicates.")
     # Compare column name and column data types
-    if existing_df.dtypes.to_dict() != delta_df.dtypes.to_dict():
+    col_diff = set(zip(existing_schema.names, existing_schema.types)) - set(zip(delta_schema.names, delta_schema.types))
+    if len(col_diff) > 0:
```

Comment:
@igorborgest I am obtaining the Arrow schema from each Pandas dataframe and comparing them so that we use the same types. Is this ok or can you think of a better way?